### DISCUSSION TITLE
NEC PC Engine 6 to 2 butons

### DISCUSSION THREAD

**Discussion by pkerga** [darthcloud/BlueRetro#252] [Ideas]
On the PC Engine's original 6-button Avenue pro pad, you can switch to 2-button mode (mode a or b) with a simple gesture.  Would this be possible (in the same way that we go from a simple controller to a dual shock on ps1)?  Indeed, some games seem to act strangely with the 6-button controller.  Thank you !

**Comment by darthcloud** [comment]
Switch back the config to regular gamepad via the web config?

Any game with no 6btns support (99%) will be broken in that mode.

**Reply by pkerga** [reply]
not always easy to get my phone back from my children 😂

**Reply by darthcloud** [reply]
You can't make that switch live btw, you need to reboot / power cycle the adapter after changing that setting btw.

**Comment by pmgducati** [comment]
Just a thought, @darthcloud is there a free GPIO pin that could be used as a ALT toggle/reboot function.  Then users could hook a button to that input and toggles to alt modes for controllers.  This could also be a good idea to be able to toggle multitap

**Reply by darthcloud** [reply]
Not sure I understand, just press ESP EN button to reboot?

**Reply by pmgducati** [reply]
What I'm suggesting is a GPIO pin that a button could be tied to that
performs the multitap or ALT controller mode toggle without having to go
into the Web page and then one successful. If it could send a reboot
command you could do it all in one button press.

I will use IO14 as an example:
![image](https://user-images.githubusercontent.com/20894227/163482894-19338fa7-78ed-4cee-82c2-da92270e18d6.png)


If IO14 goes high, toggle alt mode then reboot

I think the option for this would be great to toggle multitap also.

**Reply by darthcloud** [reply]
No new pin or button sorry, IMHO this is not needed.

If you swap the controller type often enough to find it annoying to go in the config, that tell me you are zapping through ROMs and not really playing :P .

**Reply by pkerga** [reply]
All of this is true.  I find it hard to explain the joy I feel at switching from a classic controller to a dual shock on my PSIO games list.  Strange, don't you think? 🧐😂

**Reply by darthcloud** [reply]
Is there a problem with PSIO?

**Reply by pkerga** [reply]
No no, everything works perfectly!

**Comment by pceslayer** [comment]
Would be nice to map those functions to a button much like you did with the
N64 rumble/mempack functionality. Especially if using a standalone firmware.
